src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTriggerListener.cs

```diff
@@ -148,5 +177,235 @@ private void InitializeBuilder()
                     .WithObserverFactory(this);
             }
         }
+
+        async Task<ScaleMetrics> IScaleMonitor.GetMetricsAsync()
+        {
+            return await GetMetricsAsync();
+        }
+
+        internal async Task<CosmosDBTriggerMetrics> GetMetricsAsync(IRemainingWorkEstimator workEstimator)
+        {
+            _workEstimator = workEstimator;
+            return await GetMetricsAsync();
+        }
+
+        public async Task<CosmosDBTriggerMetrics> GetMetricsAsync()
+        {
+            int partitionCount = 0;
+            long remainingWork = 0;
+            IReadOnlyList<RemainingPartitionWork> workPerPartition = null;
+
+            try
+            {
+                InitializeBuilder();
+
+                if (_workEstimator == null)
+                {
+                    _workEstimator = await this._hostBuilder.BuildEstimatorAsync();
+                }
+
+                workPerPartition = await this._workEstimator.GetEstimatedRemainingWorkPerPartitionAsync();
+
+                if (workPerPartition != null)
+                {
+                    partitionCount = workPerPartition.Count;
+                    remainingWork = workPerPartition.Sum(item => item.RemainingWork);
+                }
+            }
+            catch (Microsoft.Azure.Documents.DocumentClientException e)
+            {
+                if (!TryHandleDocumentClientException(e))
+                {
+                    _logger.LogWarning("Unable to handle Microsoft.Azure.Documents.DocumentClientException: {0}", e.Message);
+                }
+            }
+            catch (NullReferenceException e)
```

Comment:
I don't think we should be swallowing null refs

Reply:
removed